checks.d/mysql.py

```diff
@@ -32,33 +28,246 @@
+    'Innodb_pending_checkpoint_writes': ('mysql.innodb.pending_checkpoint_writes', RATE),
+    'Innodb_pending_ibuf_aio_reads': ('mysql.innodb.pending_ibuf_aio_reads', RATE),
+    'Innodb_pending_log_flushes': ('mysql.innodb.pending_log_flushes', RATE),
+    'Innodb_pending_log_writes': ('mysql.innodb.pending_log_writes', RATE),
+    'Innodb_pending_normal_aio_reads': ('mysql.innodb.pending_normal_aio_reads', RATE),
+    'Innodb_pending_normal_aio_writes': ('mysql.innodb.pending_normal_aio_writes', RATE),
+    'Innodb_queries_inside': ('mysql.innodb.queries_inside', RATE),
+    'Innodb_queries_queued': ('mysql.innodb.queries_queued', RATE),
+    'Innodb_read_views': ('mysql.innodb.read_views', GAUGE),
+    'Innodb_rows_deleted': ('mysql.innodb.rows_deleted', RATE),
+    'Innodb_rows_inserted': ('mysql.innodb.rows_inserted', RATE),
+    'Innodb_rows_read': ('mysql.innodb.rows_read', RATE),
+    'Innodb_rows_updated': ('mysql.innodb.rows_updated', RATE),
+    'Innodb_s_lock_os_waits': ('mysql.innodb.s_lock_os_waits', RATE),
+    'Innodb_s_lock_spin_rounds': ('mysql.innodb.s_lock_spin_rounds', RATE),
+    'Innodb_s_lock_spin_waits': ('mysql.innodb.s_lock_spin_waits', RATE),
+    'Innodb_semaphore_wait_time': ('mysql.innodb.semaphore_wait_time', GAUGE),
+    'Innodb_semaphore_waits': ('mysql.innodb.semaphore_waits', GAUGE),
+    'Innodb_tables_in_use': ('mysql.innodb.tables_in_use', GAUGE),
+    'Innodb_x_lock_os_waits': ('mysql.innodb.x_lock_os_waits', RATE),
+    'Innodb_x_lock_spin_rounds': ('mysql.innodb.x_lock_spin_rounds', RATE),
+    'Innodb_x_lock_spin_waits': ('mysql.innodb.x_lock_spin_waits', RATE),
+}
+
+GALERA_VARS = {
+    'wsrep_cluster_size': ('mysql.galera.wsrep_cluster_size', GAUGE),
+    'wsrep_local_recv_queue_avg': ('mysql.galera.wsrep_local_recv_queue_avg', GAUGE),
+    'wsrep_flow_control_paused': ('mysql.galera.wsrep_flow_control_paused', GAUGE),
+    'wsrep_cert_deps_distance': ('mysql.galera.wsrep_cert_deps_distance', GAUGE),
+    'wsrep_local_send_queue_avg': ('mysql.galera.wsrep_local_send_queue_avg', GAUGE),
+}
+
+PERFORMANCE_VARS = {
+    'query_run_time_avg' : ('mysql.performance.query_run_time.avg', GAUGE),
+    'perf_digest_95th_percentile_avg_us': ('mysql.performance.digest_95th_percentile.avg_us', GAUGE),
+}
+
+SCHEMA_VARS = {
+    'information_schema_size': ('mysql.info.schema.size', GAUGE),
+}
+
+REPLICA_VARS = {
+    'Seconds_Behind_Master': ('mysql.replication.seconds_behind_master', GAUGE),
+}
 
 class MySql(AgentCheck):
     SERVICE_CHECK_NAME = 'mysql.can_connect'
+    SLAVE_SERVICE_CHECK_NAME = 'mysql.replication.slave_running'
     MAX_CUSTOM_QUERIES = 20
-    DEFAULT_TIMEOUT = 5
 
     def __init__(self, name, init_config, agentConfig, instances=None):
         AgentCheck.__init__(self, name, init_config, agentConfig, instances)
         self.mysql_version = {}
         self.greater_502 = {}
+        self.host = None
```

Comment:
You shouldn't store instance level variables at the AgentCheck level. You might end up breaking multi instances configuration